.github/workflows/ci.yaml

```diff
@@ -0,0 +1,513 @@
+      NVD_API_KEY: ${{ secrets.NVD_API_KEY }}
+    runs-on: ubuntu-latest
+    needs:
+      - pre-build-updater
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: updater-build
+
+      - name: Unpack updater build
+        run: |
+          tar xvzf updater-build.tgz
+
+      - name: genesis-dump
+        run: |
+          source ./scripts/ci/lib.sh
+          generate_genesis_dump
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: genesis-dump
+          path: /tmp/genesis-dump/genesis-dump.zip
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: vuln-dump
+          path: /tmp/vuln-dump
+
+  generate-db-dump:
+    if: |
+      github.event_name != 'pull_request' ||
+      contains(github.event.pull_request.labels.*.name, 'generate-dumps-on-pr')
+    runs-on: ubuntu-latest
+    needs:
+      - generate-genesis-dump
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: updater-build
+
+      - name: Unpack updater build
+        run: |
+          tar xvzf updater-build.tgz
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: genesis-dump
+          path: /tmp/genesis-dump
+
+      - name: db-dump
+        run: |
+          source ./scripts/ci/lib.sh
+          generate_db_dump
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: db-dump
+          path: /tmp/postgres/pg-definitions.sql.gz
+
+  generate-scanner-bundle:
+    runs-on: ubuntu-latest
+    needs:
+      - pre-build-scanner
+      - generate-genesis-dump
+    if: |
+      always() &&
+      (needs.generate-genesis-dump.result == 'success' || needs.generate-genesis-dump.result == 'skipped')
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-build
+
+      - name: Unpack scanner build
+        run: |
+          tar xvzf scanner-build.tgz
+
+      - uses: actions/download-artifact@v3
+        if: |
+          github.event_name != 'pull_request' ||
+          contains(github.event.pull_request.labels.*.name, 'generate-dumps-on-pr')
```

Comment:
Could also be `needs.generate-genesis-dump.result == 'success'` maybe? Probably doesn't matter that much